lib/logmsg/tests/test_log_message.c

```diff
@@ -382,3 +382,68 @@ Test(log_message, test_sdata_value_omits_unset_values)
   assert_sdata_value_equals(msg, "");
   log_msg_unref(msg);
 }
+
+#define DEFUN_KEY_VALUE(name, key, value, size) \
+  gchar name ## _key[size]; \
+  gchar name ## _value[size]; \
+  name ## _key[size-1] = name ## _value[size-1] = 0; \
+  memset(name ## _key, key, sizeof(name ##_key)-1); \
+  memset(name ## _value, value, sizeof(name ##_value)-1); \
+
+
+typedef struct
+{
+  guint32 nvtable_size_old;
+  guint32 nvtable_size_new;
+  guint32 msg_size_old;
+  guint32 msg_size_new;
+} sizes_t;
+
+sizes_t
+add_kv_and_check(LogMessage *msg, gchar *key, gchar *value)
+{
+  sizes_t sizes;
+  sizes.msg_size_old = log_msg_get_size(msg);
+  sizes.nvtable_size_old = nv_table_get_memory_consumption(msg->payload);
+  log_msg_set_value_by_name(msg, key, value, -1);
+  sizes.nvtable_size_new = nv_table_get_memory_consumption(msg->payload);
+  sizes.msg_size_new = log_msg_get_size(msg);
+  return sizes;
+}
+
+Test(log_message, test_message_size)
+{
+  LogMessage *msg = log_msg_new_empty();
+
+  sizes_t sizes;
+  DEFUN_KEY_VALUE(small, 'C', 'D', 10);
+  sizes = add_kv_and_check(msg, small_key, small_value);
+  // 9*'C'+'\0' + 9*'D'+'\0'
+  guint32 entry_size = NV_ENTRY_DIRECT_HDR + 9 + 9 + 2;
```

Comment:
hmmm.... isn't this fragile?
I mean... if someone will change `9` to for exmaple... `12`, then it will fail...
We can keep this, but this code is depending on a very internal detail of the nvtable implementation, as I see.